app/assets/javascripts/alchemy/alchemy.sitemap.js.coffee

```diff
@@ -4,12 +4,51 @@ window.Alchemy = {} if typeof(window.Alchemy) is 'undefined'
 Alchemy.Sitemap =
 
   # Storing some objects.
-  init: ->
+  init: (url, page_root_id, full, callback) ->
     @search_field = $("#search_field")
     @filter_field_clear = $('.js_filter_field_clear')
     @display = $('#page_filter_result')
-    @items = $(".sitemap_page", '#sitemap')
-    @_observe()
+    @sitemap_wrapper = $('#sitemap-wrapper')
+    @template = Handlebars.compile($('#sitemap-template').html())
+    list_template_regexp = new RegExp '\/' + page_root_id, 'g'
+    @list_template = $('#sitemap-list').html().replace(list_template_regexp, '/{{id}}')
+    @items = null
+    @url = url
+    @page_root_id = page_root_id
+    @full = full
+    @callback = callback
+    @fetch()
+
+  # Fetches the sitemap from JSON
+  fetch: (foldingId) ->
+    self = Alchemy.Sitemap
+    request = $.ajax url: @url, data:
+      id: @page_root_id
+      full: @full
+    spinner = Alchemy.Spinner.small()
+
+    if foldingId
+      spinTarget = $('#fold_button_' + foldingId)
+    else
+      spinTarget = self.sitemap_wrapper.find('.sitemap-load-notice')
+
+    spinner.spin(spinTarget[0])
+
+    request.done (data) ->
+      Handlebars.registerPartial('list', self.list_template)
+      # This will also remove the spinner
+      self.sitemap_wrapper.html(self.template({children: data.pages}))
+      self.items = $(".sitemap_page", '#sitemap')
+      self._observe()
+
+      Alchemy.watchForDialogs('#sitemap')
+
+      if self.callback
+        self.callback()
+
+    # TODO: Prettify this.
+    request.fail (jqXHR, status) ->
+      alert("Request failed: " + status)
```

Comment:
We have `Alchemy.debug` that [provides pretty console output](https://github.com/AlchemyCMS/alchemy_cms/blob/master/app/assets/javascripts/alchemy/alchemy.base.js.coffee#L116) for logging errors like this.